typesafety/test_methods/test_cond.yml

```diff
@@ -0,0 +1,62 @@
+- case: cond_result
+  disable_cache: true
+  main: |
+    from returns.methods import cond
+    from returns.result import Result
+
+    reveal_type(cond(Result, True, 42, '42'))  # N: Revealed type is 'returns.result.Result*[builtins.int*, builtins.str*]'
+
+- case: cond_ioresult
+  disable_cache: true
+  main: |
+    from returns.io import IOResult
+    from returns.methods import cond
+
+    reveal_type(cond(IOResult, False, 'success', 'failure'))  # N: Revealed type is 'returns.io.IOResult*[builtins.str*, builtins.str*]'
+
+- case: cond_future_result
+  disable_cache: true
+  main: |
+    from returns.future import FutureResult
+    from returns.methods import cond
+
+    reveal_type(cond(FutureResult, False, True, False))  # N: Revealed type is 'returns.future.FutureResult*[builtins.bool*, builtins.bool*]'
+
+- case: cond_reader_ioresult
+  disable_cache: true
+  main: |
+    from returns.methods import cond
+    from returns.context import ReaderIOResult
+
+    reveal_type(cond(ReaderIOResult, True, 1.0, False))  # N: Revealed type is 'returns.context.requires_context_ioresult.RequiresContextIOResult*[builtins.float*, builtins.bool*, <nothing>]'
+
+- case: cond_reader_future_result
+  disable_cache: true
+  main: |
+    from returns.methods import cond
+    from returns.context import ReaderFutureResult
+
+    reveal_type(cond(ReaderFutureResult, True, 1, 1.0))  # N: Revealed type is 'returns.context.requires_context_future_result.RequiresContextFutureResult*[builtins.int*, builtins.float*, <nothing>]'
```

Comment:
Should be `returns.context.requires_context_future_result.RequiresContextFutureResult*[builtins.int*, builtins.float*, Any]`

Reply:
I think we have a `type` problem here or not!! It just work if I assign the result to a variable and typing it properly:
```yml
- case: cond_reader_ioresult
  disable_cache: true
  main: |
    from typing import Any

    from returns.methods import cond
    from returns.context import ReaderIOResult

    container: ReaderIOResult[float, bool, Any] = cond(ReaderIOResult, True, 1.0, False)
    reveal_type(container)  # N: Revealed type is 'returns.context.requires_context_ioresult.RequiresContextIOResult[builtins.float, builtins.bool, Any]'
```

Reply:
Sorry, it's working without the above solution!! It was `mypy` cache kidding me hahaha
I just deleted the cache an everything worked properly!!